sdk/core/azure_core/src/http/pager.rs

```diff
@@ -811,14 +811,36 @@ impl<P> fmt::Debug for PageIterator<P> {
     }
 }
 
-#[derive(Debug, Clone, Eq)]
-enum State<C> {
+#[derive(Clone, Eq)]
+enum State<C>
+where
+    C: AsRef<str>,
+{
     Init,
     More(C),
     Done,
 }
 
-impl<C> PartialEq for State<C> {
+impl<C> fmt::Debug for State<C>
+where
+    C: AsRef<str>,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            State::Init => write!(f, "State::Init"),
```

Comment:
Nit: default formatting only uses the variant name if you want to match e.g., "Init" not "State::Init".